src/pytorch_lightning/core/module.py

```diff
@@ -559,8 +561,19 @@ def log_dict(
                 would produce a deadlock as not all processes would perform this log call.
         """
         if self._fabric is not None:
-            return self._log_dict_through_fabric(dictionary=dictionary, logger=logger)
-        for k, v in dictionary.items():
+            return self._log_dict_through_fabric(dictionary=dictionary, logger=logger)  # type: ignore[arg-type]
+
+        kwargs: Dict[str, bool] = {}
+
+        if isinstance(dictionary, MetricCollection):
+            kwargs["keep_base"] = False
+            if (
+                "copy_state" in inspect.signature(MetricCollection.items).parameters
```

Comment:
I wondered if using `RequirementCache("torchmetrics<=...")` instead of `compare_version` would have worked too.

Reply:
Looks like it worked.